twilight-gateway/src/shard.rs

```diff
@@ -5,72 +5,21 @@
 //! maintaining an identified session with the Discord gateway. For more
 //! information about what a shard is in the context of Discord's gateway API,
 //! refer to the documentation for [`Shard`].
-//!
-//! # Implementation Flow
-//!
-//! Other than informative methods and getters, shards have two functions:
-//! sending websocket messages and receiving websocket messages, called
-//! "events" in the context of the Discord gateway.
-//!
-//! Sending a message is simple and the flow of it looks like:
-//!
-//! 1. If the user sends a [command] via [`Shard::command`] the command is
-//! serialized into a raw websocket message and then...
-//! 2. [`Shard::send`] is called and the sending of the message goes through
-//! ratelimiting via [`CommandRatelimiter`] if ratelimiting [is enabled];
-//! 3. The [websocket message] is sent over the [websocket connection].
-//!
-//! Receiving a message is a little bit more complicated, but follows as:
-//!
-//! 1. The user calls [`Shard::next_message`] to receive the next message from
-//! the Websocket;
-//! 2. If the shard is disconnected then the connection is reconnected;
-//! 3. One of five things wait to happen:
-//!   a. the shard receives a close frame from the user over the [user channel],
-//!   which is then forwarded via [`Shard::close`]; or
-//!   b. the interval for the shard to send the next heartbeat occurs, in which
-//!   case [`Shard::heartbeat`] is called; or
-//!   c. the identify receiver is ready, in which case [`Shard::send`] is called
-//!   with the identify payload; or
-//!   d. the shard receives a command from the user over the [user channel],
-//!   which is then forwarded via [`Shard::send`]; or
-//!   e. the shard receives a message from Discord via the websocket connection.
-//! 4. In the case of 3(a) through 3(d), 3 is repeated; otherwise...
-//! 5. If the message is not a close it's [processed] by the shard;
-//! 6. The raw Websocket message is returned to the user
-//!
-//! If the user called [`Shard::next_event`] instead of [`Shard::next_message`],
```

Comment:
I think the top-level documentation should at least mention how to get an event instead of a message, yet another example would be a bit much

Reply:
Dokumentation can contain bugs as well as code (like here, where the documentation was not kept in sync) and I believe the code speaks for itself. If not, the *code* readability should be improved instead of adding documentation. Retrieving an event should already be covered by the `Shard` documentation.

Reply:
> If not, the *code* readability should be improved instead of adding documentation
Users shouldn't need to read the code of the methods they are calling. But yeah the shard documentation does "cover" this, even though it doesn't explicitly mention it in the text.